qiskit/tools/parallel.py

```diff
@@ -51,12 +51,18 @@
 """
 
 import os
+import sys
 import platform
-from concurrent.futures import ProcessPoolExecutor
+import multiprocessing
 from qiskit.exceptions import QiskitError
 from qiskit.util import local_hardware_info
 from qiskit.tools.events.pubsub import Publisher
 
+if sys.platform == 'darwin':
+    POOL = multiprocessing.get_context('fork').Pool
```

Comment:
Wasn't there an issue with fork on macOS which is why they changed the default behavior in 3.8 (which prompted #3491 in the first place)?